### DISCUSSION TITLE
Using pyrfc to fetch data from SAP (UUID Issue)

### DISCUSSION THREAD

**Discussion by YashGhaisas** [SAP/PyRFC#256] [Q&A] [answered]
Hello everyone, 
I am using pyrfc library to fetch the data from SAP. 
I have a column named UUID in SAP table which is varchar(32). When I get the data in python using RFC_READ_TABLE, only 16 characters are getting fetched rather than all 32. 
For example - 
If the value is - '005056AE436A1EE682FA84AD95D5B572', then I'm getting only - '005056AE436A1EE6'

I have mentioned the code below, please check 

**#####CODE#####**

def main():
    
    try:
        conn = Connection(ashost=ASHOST, sysnr=SYSNR, client=CLIENT, user=USER, passwd=PASSWD)
        pp = PrettyPrinter(indent=5)
        ROWS_AT_A_TIME = 100 
        rowskips = 0
        size = 100
        FIELD = ['ADRP_UUID']
        df_all = pd.DataFrame()
        while True:
            print("----Begin of Batch---")
            result = conn.call('RFC_READ_TABLE', QUERY_TABLE = 'ADRP', DELIMITER = '|', FIELDS = FIELD,  ROWSKIPS = rowskips, ROWCOUNT = ROWS_AT_A_TIME)
            #pp.pprint(result['DATA'])
            df = pd.DataFrame(result['DATA'])
            new= df["WA"].str.split("|", n= 2, expand= True)
            print(new)
            df_all = df_all.append(new)
            rowskips += ROWS_AT_A_TIME
            if len(result['DATA']) < ROWS_AT_A_TIME:
                break

    except CommunicationError:
        print("Could not connect to server.")
        raise
    except LogonError:
        print("Could not log in. Wrong credentials?")
        raise
    except (ABAPApplicationError, ABAPRuntimeError):
        print("An error occurred.")
        raise
        
    return df_all,result 

    
if __name__ == '__main__':
    main()

**Comment by bsrdjan** [comment, marked as answer]
This is one of RFC_READ_TABLE restrictions, PyRFC can't help fix that. Search for `RFC_READ_TABLE` in issues will find few similar questions answered there.
